File: apps/junior/src/api/agent.ts
// 接口文档 §3, §4:Agent 家书生成 + 对话
import { USE_REAL_API, apiRequest, mockDelay } from './client'

export type GenerateSummaryInput = {
  date: string
  user_id: string
}

export type GenerateSummaryResult = {
  summary_id: string
  title: string
  content: string // 家书正文(一整段小作文)
  cover_image: string[]
  suggested_questions: string[]
}

// POST /api/v1/agent/generate-summary
export async function generateSummary(
  input: GenerateSummaryInput,
): Promise<GenerateSummaryResult> {
  if (USE_REAL_API) {
    return apiRequest<GenerateSummaryResult>('/agent/generate-summary', {
      method: 'POST',
      body: JSON.stringify(input),
    })
  }

  // Mock:返回固定小作文(5 句),summary_id 用时间戳保证唯一
  return mockDelay(
    {
      summary_id: `sum_${Date.now()}`,
      title: '小明在南山散步的一天',
      content:
        '今天下午五点多，小明和同事走到南山脚下散步，秋风吹得人很舒服。' +
        '走到山腰的观景台时，城里的灯正一盏一盏亮起来，远远看像一条发光的带子。' +
        '晚饭他和朋友去了山下的老店，点了一锅热气腾腾的火锅，两个人一直聊到很晚。' +
        '回家路上他经过一个水果摊，顺手买了一袋橘子，说是明天带去办公室分给大家。' +
        '妈妈，他今天过得挺好的，您放心。',
      cover_image: [],
      suggested_questions: [
        '今天和谁一起吃的火锅呀？',
        '最近工作忙不忙，身体还好吗？',
      ],
    },
    400,
  )
}

export type ChatInput = {
  query: string
  summary_id: string
}

export type ChatResult = {
  reply_text: string
  action: 'reply' | 'notify_younger'
  message_id?: string // 若 action=notify_younger,后端落库后返回消息 ID(扩展字段)
}

// POST /api/v1/agent/chat
export async function chatWithAgent(input: ChatInput): Promise<ChatResult> {
  if (USE_REAL_API) {
    return apiRequest<ChatResult>('/agent/chat', {
      method: 'POST',
      body: JSON.stringify(input),
    })
  }

  // Mock:简单拼接用户问的话
  const q = input.query.trim()
  if (!q) {
    return mockDelay({
      reply_text: '妈妈，我刚才没听清您说什么，您再说一遍好吗？',
      action: 'reply' as const,
    })
  }
  return mockDelay(
    {
      reply_text: `妈妈，我刚才听到您说，${q}。等小明回来，我把这个问题转告他，让他再写一封信给您。`,
      action: 'reply' as const,
    },
    500,
  )
}

File: backend/app/api/agent.py
import json
from datetime import datetime, timedelta
from uuid import uuid4

from fastapi import APIRouter, Depends
from fastapi.responses import JSONResponse
from sqlalchemy.orm import Session

from app.deps import get_agent, get_db
from app.models import Diary, Message, RawData
from app.schemas.requests import ChatRequest, GenerateSummaryRequest
from app.schemas.responses import ApiResponse, ChatReplyData, DiaryData
from app.services.agent import FamLinkAgent

router = APIRouter()


@router.post("/generate-summary", response_model=ApiResponse[DiaryData])
def generate_summary(
    req: GenerateSummaryRequest,
    db: Session = Depends(get_db),
    agent: FamLinkAgent = Depends(get_agent),
):
    try:
        start = datetime.strptime(req.date, "%Y-%m-%d")
    except ValueError:
        return JSONResponse(
            status_code=400,
            content={"code": 400, "message": "日期格式错误", "data": None},
        )
    end = start + timedelta(days=1)

    rows = (
        db.query(RawData)
        .filter(
            RawData.status == "pending",
            RawData.created_at >= start,
            RawData.created_at < end,
        )
        .order_by(RawData.created_at)
        .all()
    )
    if not rows:
        return JSONResponse(
            status_code=400,
            content={
                "code": 400,
                "message": "当日没有可处理的数据",
                "data": None,
            },
        )

    raw_items = [
        {"type": r.type, "content": r.content, "description": r.description}
        for r in rows
    ]
    result = agent.generate_diary(raw_items, req.date)
    summary_id = f"sum_{uuid4().hex[:8]}"

    diary = Diary(
        summary_id=summary_id,
        date=req.date,
        title=result["title"],
        content=result["content"],
        cover_image=json.dumps(result["cover_image"], ensure_ascii=False),
        suggested_questions=json.dumps(result["suggested_questions"], ensure_ascii=False),
        raw_data_ids=json.dumps([r.item_id for r in rows], ensure_ascii=False),
    )
    db.add(diary)
    for r in rows:
        r.status = "processed"
    db.commit()

    return ApiResponse[DiaryData](
        code=200,
        message="Agent 处理完成",
        data=DiaryData(
            summary_id=summary_id,
            title=result["title"],
            content=result["content"],
            cover_image=result["cover_image"],
            suggested_questions=result["suggested_questions"],
        ),
    )


@router.post("/chat", response_model=ApiResponse[ChatReplyData])
def chat(
    req: ChatRequest,
    db: Session = Depends(get_db),
    agent: FamLinkAgent = Depends(get_agent),
):
    diary = db.query(Diary).filter_by(summary_id=req.summary_id).first()
    if diary is None:
        return JSONResponse(
            status_code=404,
            content={"code": 404, "message": "日记不存在", "data": None},
        )

    chat_result = agent.chat_with_elder(
        diary_title=diary.title,
        diary_content=diary.content,
        elder_query=req.query,
    )

    if chat_result["action"] == "reply":
        return ApiResponse[ChatReplyData](
            code=200,
            message="success",
            data=ChatReplyData(
                reply_text=chat_result["reply_text"],
                action="reply",
            ),
        )

    transfer_result = agent.transfer_message(
        summary_id=req.summary_id,
        elder_query=req.query,
        agent_reply=chat_result["reply_text"],
    )
    msg = Message(
        summary_id=req.summary_id,
        elder_query=req.query,
        agent_reply=chat_result["reply_text"],
        transfer_content=transfer_result["transfer_content"],
        is_transferred=False,
    )
    db.add(msg)
    db.commit()

    return ApiResponse[ChatReplyData](
        code=200,
        message="success",
        data=ChatReplyData(
            reply_text=chat_result["reply_text"],
            action="notify_younger",
            transfer_content=transfer_result["transfer_content"],
        ),
    )
